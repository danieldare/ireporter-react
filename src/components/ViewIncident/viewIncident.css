.records {
  display: flex;
  justify-content: space-evenly;
  margin-bottom: 200px;
  flex-wrap: wrap;
}

.one-record {
  padding: 10px 15px;
  margin: 0 0 50px 0;
  border-radius: 3px;
  border: 1px solid #ccc;
  display: inline-block;
  min-width: 350px;
  border-left: 3px solid #3c4859;
  overflow: hidden;
  position: relative;
  margin-right: 13px;
  animation: fadeout 2s;
}

.one-record__title-response--small {
  font-size: 1.8rem;
  font-weight: 600;
  font-family: "Nunito", sans-serif;
}

.one-record__title-response {
  font-size: 2.1rem;
  font-weight: 300;
  font-family: "Nunito", sans-serif;
  text-transform: capitalize;
  padding: 10px 0 0 0;
}

.one-response__type--small {
  font-size: 1rem;
  padding: 2px 7px;
  background: #d1603d;
  color: #fff;
  letter-spacing: 0.1rem;
  display: inline-block;
  border-radius: 6px;
  font-weight: 600;
  text-transform: uppercase;
}

.one-response__type {
  font-size: 1.8rem;
  padding: 2px 7px;
  background: #d1603d;
  color: #fff;
  letter-spacing: 0.1rem;
  display: inline-block;
  border-radius: 6px;
  font-weight: 600;
  text-transform: uppercase;
}

.one-record__view-detail {
  padding: 5px 15px;
  background: #3c4859;
  color: #fff;
  text-decoration: none;
  font-size: 1.3rem;
  border-radius: 4px;
  font-weight: 100;
  font-family: "Open Sans", sans-serif;
  display: inline-block;
}

.one-record__title--small {
  font-weight: 300;
  font-family: "Open Sans", sans-serif;
}

.one-record__title {
  font-weight: 600;
  font-size: 1.5rem;
  border-bottom: 2px solid #d1603d;
  color: #3c4859;

  /* font-family: "Open Sans", sans-serif; */
}

.tick {
  width: 41px;
  display: block;
  text-align: center;
  margin: 0 auto;
}

.tick--small {
  width: 21px;
  display: block;
  text-align: center;
  margin: 0 auto;
}

.one-record__location {
  font-size: 1rem;
  font-weight: 500;
  text-transform: lowercase;
}

.one-record__time {
  font-size: 1.3rem;
  text-transform: initial;
  font-weight: 500;
  font-family: "Open Sans", sans-serif;
}

.one-record__footer {
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.record-title {
  font-size: 2.5rem;
  font-weight: 300;
  font-family: "Nunito", sans-serif;
  margin: 30px 0;
  border-left: 2px solid #d1603d;
  padding: 0 10px;

  border-radius: 3px;
}

.one-record__status-response {
  text-align: center;
  text-transform: initial;
  margin: 30px auto;
  font-weight: 300;
  font-family: "Open Sans", sans-serif;
  font-size: 3.5rem;
  text-transform: uppercase;
}

.one-record__status-response--small {
  text-align: center;
  text-transform: initial;
  margin: 30px auto;
  text-transform: uppercase;
  font-weight: 800;
  font-family: "Nunito", sans-serif;
  font-size: 2.5rem;
}
