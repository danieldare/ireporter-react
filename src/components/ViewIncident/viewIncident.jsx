import React, { useEffect } from 'react';
import { Link, withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import dayjs from 'dayjs';
import { fetchIncident } from '../../actions/actionCreators/viewIncident';

const viewIncident = props => {
  const {
    match: { url }
  } = props;
  useEffect(() => {
    if (url === '/view-redflag') {
      props.fetchIncident('red-flags');
    }
    if (url === '/view-intervention') {
      props.fetchIncident('interventions');
    }
  }, []);

  const {
    Records: { incidents, isFetching }
  } = props;

  let records = '';

  if (isFetching) {
    records = <div className="spinner-2" />;
  }

  if (!isFetching) {
    records = incidents
      .sort((a, b) => (a.id < b.id ? 1 : -1))
      .map(record => {
        let img = '';
        if (record.status === 'draft')
          img =
            'https://res.cloudinary.com/decqfpglp/image/upload/v1556488892/envelope.png';
        else if (record.status === 'rejected')
          img =
            'https://res.cloudinary.com/decqfpglp/image/upload/v1551650056/cancel.png';
        else if (record.status === 'resolved')
          img =
            'https://res.cloudinary.com/decqfpglp/image/upload/v1551647685/checked.png';
        else if (record.status === 'under-investigation')
          img =
            'https://res.cloudinary.com/decqfpglp/image/upload/v1551650337/Rolling-5s-200px.gif';

        return (
          <div className="one-record" key={record.createdon}>
            <div className="left-border__record">
              <small className="one-record__title--small">Title</small>
              <p className="one-record__title-response--small">
                {record.title}
              </p>
              <p className="one-response__type--small">{record.type}</p>
            </div>
            <div className="one-record__status-response--small">
              <img src={img} className="tick--small" alt="img" />
              {record.status}
              <p className="one-record__location">
                Location: {record.location}
              </p>
            </div>

            <div className="one-record__footer">
              <p className="one-record__time">
                {dayjs(record.createdon)
                  .add(1, 'day')
                  .subtract(1, 'year')
                  .toString()}
              </p>
              <Link
                to={
                  url === '/view-redflag'
                    ? `/view-single-redflag?id=${record.id}`
                    : `/view-single-intervention?id=${record.id}`
                }
                className="one-record__view-detail"
              >
                view details
              </Link>
            </div>
            <img
              src="https://res.cloudinary.com/decqfpglp/image/upload/v1550621617/warning.png"
              className="view__img"
              alt="img"
            />
          </div>
        );
      });
  }

  if (!isFetching && !incidents.length) records = <span>NO RESULTS FOUND</span>;

  return (
    <div className="container">
      <p className="record-title">
        {url === '/view-intervention'
          ? 'Intervention Records'
          : 'Redflag Records'}
      </p>
      <div className="records">{records}</div>
    </div>
  );
};

const mapStateToProps = state => {
  return {
    Records: state.Records
  };
};

export default connect(
  mapStateToProps,
  { fetchIncident }
)(withRouter(viewIncident));
