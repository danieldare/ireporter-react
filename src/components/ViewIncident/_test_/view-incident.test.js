import React from 'react';
import { Provider } from 'react-redux';
import { mount } from 'enzyme';
import { MemoryRouter as Router } from 'react-router-dom';
import ViewIncident from '../viewIncident';
import { mockStore } from '../../../../test/setupTests.js';

const props = {
  Records: {
    Records: 'records',
    incidents: []
  }
};

describe('<ViewIncident /> rendering', () => {
  it('should render without crashing', () => {
    const store = mockStore(props);
    const wrapper = mount(
      <Provider store={store}>
        <Router>
          <ViewIncident />
        </Router>
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
