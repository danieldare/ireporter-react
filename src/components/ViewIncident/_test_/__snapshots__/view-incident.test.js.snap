// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ViewIncident /> rendering should render without crashing 1`] = `ReactWrapper {}`;
